### ISSUE TITLE
Adding Shellcheck to Scoop

### ISSUE THREAD

**Issue by Calinou** [lukesampson/scoop#1920]
Based on what [`godot.json`](https://github.com/lukesampson/scoop-extras/blob/master/godot.json) does, I attempted adding Shellcheck to Scoop using this descriptor file:

```json
{
    "homepage": "https://shellcheck.net/",
    "version": "0.4.7",
    "license": "GPL-3.0",
    "url": "https://shellcheck.storage.googleapis.com/shellcheck-v0.4.7.zip",
    "hash": "bbc260e15b42efdce628222d5649c0fa4cb3adaf2f71c958580b8928fe039e06",
    "bin": "shellcheck.exe",
    "pre_install": "Rename-Item $dir\\shellcheck-*.exe $dir\\shellcheck.exe"
}
```

Here is the install log:

```
Installing 'shellcheck' (0.4.7) [64bit]
Loading shellcheck-v0.4.7.zip from cache
Checking hash of shellcheck-v0.4.7.zip... ok.
Extracting... done.
Running pre-install script...
Rename-Item : Cannot process argument because the value of argument "path" is not valid. Change the value of the "path" argument and run the operation again.
At line:1 char:1
+ Rename-Item $dir\shellcheck-*.exe $dir\shellcheck.exe
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Rename-Item], PSArgumentException
    + FullyQualifiedErrorId : Argument,Microsoft.PowerShell.Commands.RenameItemCommand

Linking ~\scoop\apps\shellcheck\current => ~\scoop\apps\shellcheck\0.4.7
Creating shim for 'shellcheck'.
Can't shim 'shellcheck.exe': File doesn't exist.
```

The linked ZIP archive contains the executable "directly" (without any sub-directories), and the executable name changes based on the version (like Godot, which installs successfully using Scoop on my machine). I've also tried using `shellcheck-*.exe` instead of `$dir\\shellcheck-*.exe` in the `Rename-Item` command, to no avail.

**Comment by rasa** [comment]
This worked for me:
````
{
    "homepage": "https://shellcheck.net/",
    "version": "0.4.7",
    "license": "GPL-3.0",
    "url": "https://shellcheck.storage.googleapis.com/shellcheck-v0.4.7.zip",
    "hash": "bbc260e15b42efdce628222d5649c0fa4cb3adaf2f71c958580b8928fe039e06",
    "bin": "shellcheck.exe",
    "pre_install": "Get-ChildItem \"$dir\\shellcheck-*.exe\" | Rename-Item -NewName \"$dir\\shellcheck.exe\""
}
````

**Comment by rasa** [comment]
I tried adding
````
    "checkver": {
        "url": "https://shellcheck.storage.googleapis.com/index.html",
        "re": "/shellcheck-v([\\d.]+)\\.zip",
        "reverse": true
    },
    "autoupdate": {
        "url": "https://shellcheck.storage.googleapis.com/shellcheck-v$version.zip"
    }
````
but the page https://shellcheck.storage.googleapis.com/index.html is generated dynamically via javascript.

**Comment by r15ch13** [comment]
@rasa then remove `index.html` and you get this lovely XML document: https://shellcheck.storage.googleapis.com/ 😁

**Comment by rasa** [comment]
But of course!

**Comment by rasa** [comment]
OK, this works:
````
{
    "homepage": "https://shellcheck.net/",
    "version": "0.4.7",
    "license": "GPL-3.0",
    "url": "https://shellcheck.storage.googleapis.com/shellcheck-v0.4.7.zip",
    "hash": "bbc260e15b42efdce628222d5649c0fa4cb3adaf2f71c958580b8928fe039e06",
    "bin": "shellcheck.exe",
    "pre_install": "Get-ChildItem \"$dir\\shellcheck-*.exe\" | Rename-Item -NewName \"$dir\\shellcheck.exe\"",
    "checkver": {
        "url": "https://shellcheck.storage.googleapis.com/",
        "re": "<Key>shellcheck-v([\\d.]+)\\.zip</Key>",
        "reverse": true
    },
    "autoupdate": {
        "url": "https://shellcheck.storage.googleapis.com/shellcheck-v$version.zip"
    }
}
````

**Comment by rasa** [comment]
I committed the above script in https://github.com/lukesampson/scoop/pull/1922

**Comment by rasa** [comment]
Fixed in #1922
